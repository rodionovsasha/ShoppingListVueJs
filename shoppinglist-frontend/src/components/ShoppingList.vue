<template>
  <div class="col-md-6 offset-md-3">
    <error-alert v-if="error" v-bind:message="message"/>

    <h1>Shopping lists</h1>
    <div v-if="loading">Loading...</div>
    <ul class="list-unstyled" v-else>
      <li v-for="(list, index) in itemsLists" v-bind:key="list.id">
        <p>
          <router-link v-bind:to="{ path: '/itemsList/' + list.id }" class="btn btn-outline-secondary">{{ list.name }}</router-link>
          <a v-bind:href="'/#/itemsList/edit?id=' + list.id" class="btn btn-warning btn-sm">Edit</a>
          <a href="#" class="btn btn-danger btn-sm" @click.prevent="confirmDelete(list.id, index)">Delete</a>
        </p>
      </li>
    </ul>
    <router-link to="/addItemsList" class="btn btn-success btn-sm" data-toggle="tooltip" data-placement="right" title="Add new list" aria-hidden="true">
      <span class="oi oi-plus"></span>
    </router-link>
  </div>
</template>

<script>
import {AXIOS} from './http-common'
import ErrorAlert from './ErrorAlert'

export default {
  name: 'ShoppingList',
  components: {ErrorAlert},
  data () {
    return {
      itemsLists: [],
      loading: true,
      error: false,
      message: ''
    }
  },
  mounted () {
    AXIOS.get('/')
      .then(response => {
        this.itemsLists = response.data
        this.loading = false
        this.error = false
      })
      .catch(error => {
        console.log(error)
        this.error = true
        this.message = error.toString()
        this.loading = false
      })
  },
  methods: {
    confirmDelete: function (id, index) {
      this.$dialog.confirm('Are you sure you want to delete this list?')
        .then(() => {
          console.log('Delete clicked')
          AXIOS.delete('/itemsList/' + id)
            .then(() => {
              // remove list from array
              this.itemsLists.splice(index, 1)
              this.error = false
            })
            .catch(error => {
              console.log(error)
              this.error = true
              this.message = error.toString()
            })
        })
        .catch(function () {
          console.log('Cancel delete clicked')
        })
    }
  }
}
</script>
