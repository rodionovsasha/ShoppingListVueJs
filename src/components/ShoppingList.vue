<template>
  <div class="col-md-6 offset-md-3" v-if="error">
    <div class="alert alert-danger" role="alert">
      <strong>Service unavailable.</strong> Please try back later.
    </div>
  </div>
  <div class="col-md-6 offset-md-3" v-else>
    <h1>Shopping list</h1>
    <div v-if="loading">Loading...</div>
    <ul class="list-unstyled" v-else>
      <li v-for="list in itemsLists" v-bind:key="list.id">
        <p>
          <a v-bind:href="'/itemsList?id=' + list.id" class="btn btn-outline-secondary">{{ list.name }}</a>
          <a v-bind:href="'/itemsList/edit?id=' + list.id" class="btn btn-warning btn-sm">Edit</a>
          <a v-bind:href="'/itemsList/delete?id=' + list.id" class="btn btn-danger btn-sm" onclick="return confirm('Are you sure you want to delete this list?')">Delete</a>
        </p>
      </li>
    </ul>
    <a href="/itemsList/add" class="btn btn-success btn-sm" data-toggle="tooltip" data-placement="right"
       title="Add new list" aria-hidden="true">
      <span class="oi oi-plus"></span>
    </a>
  </div>
</template>

<script>
import axios from 'axios'

export default {
  name: 'ShoppingList',
  data () {
    return {
      itemsLists: [],
      loading: true,
      error: false
    }
  },
  mounted () {
    axios
      .get('http://localhost:8000/v1/api')
      .then(response => {
        this.itemsLists = response.data
        this.loading = false
      })
      .catch(error => {
        console.log(error)
        this.error = true
        this.loading = false
      })
  }
}
</script>
