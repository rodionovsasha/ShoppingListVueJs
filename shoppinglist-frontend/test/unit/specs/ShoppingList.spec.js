import Vue from 'vue'
import ShoppingList from '@/components/ShoppingList'

describe('ShoppingList.vue', () => {
  it('Should render correct heading', () => {
    const Constructor = Vue.extend(ShoppingList);
    const vm = new Constructor().$mount();
    expect(vm.$el.querySelector('h1').textContent)
      .toEqual('Shopping lists')
  })
});
