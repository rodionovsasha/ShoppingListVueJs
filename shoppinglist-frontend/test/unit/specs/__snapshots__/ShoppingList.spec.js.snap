// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShoppingList.vue Should render correct html snapshot with loading 1`] = `
<div class="col-md-6 offset-md-3">
    <!---->
    <h1>Shopping lists</h1>
    <div>Loading...</div>
    <router-link to="/addItemsList" class="btn btn-success btn-sm">Add</router-link>
</div>
`;

exports[`ShoppingList.vue Should render correct html snapshot without loading 1`] = `
<div class="col-md-6 offset-md-3">
    <!---->
    <h1>Shopping lists</h1>
    <ul class="list-unstyled">
        <li>
            <p>
                <router-link to="[object Object]" class="btn btn-outline-secondary">list1</router-link>
                <router-link to="[object Object]" class="btn btn-warning btn-sm mx-1">Edit</router-link>
                <a href="#" class="btn btn-danger btn-sm">Delete</a>
            </p>
        </li>
    </ul>
    <router-link to="/addItemsList" class="btn btn-success btn-sm">Add</router-link>
</div>
`;

exports[`ShoppingList.vue Should render error message when backend is not available 1`] = `
<div class="col-md-6 offset-md-3">
    <div>
        <div role="alert" class="alert alert-danger"><strong>Network connection</strong></div>
    </div>
    <h1>Shopping lists</h1>
    <div>Loading...</div>
    <router-link to="/addItemsList" class="btn btn-success btn-sm">Add</router-link>
</div>
`;
